Mathlib/Data/Polynomial/Splits.lean

```diff
@@ -185,11 +193,11 @@ theorem roots_ne_zero_of_splits' {f : K[X]} (hs : Splits i f) (hf0 : natDegree (
   let ⟨x, hx⟩ := exists_root_of_splits' i hs fun h => hf0 <| natDegree_eq_of_degree_eq_some h
   fun h => by
   rw [← eval_map] at hx
-  cases h.subst ((mem_roots _).2 hx)
-  exact ne_zero_of_nat_degree_gt (Nat.pos_of_ne_zero hf0)
+  have : f.map i ≠ 0 := by intro; simp_all
+  cases h.subst ((mem_roots this).2 hx)
```

Comment:
What goes wrong with the original?

Reply:
Lean can't synthesise the term `map i f ≠ 0`, which would fill the underscore.